lib/logstash/outputs/rabbitmq.rb

```diff
@@ -76,20 +71,108 @@ def initialize(params)
     super
   end
 
-  # Use MarchHare on JRuby to avoid IO#select CPU spikes
-  # (see github.com/ruby-amqp/bunny/issues/95).
-  #
-  # On MRI, use Bunny.
-  #
-  # See http://rubybunny.info and http://rubymarchhare.info
-  # for the docs.
-  if RUBY_ENGINE == "jruby"
-    require "logstash/outputs/rabbitmq/march_hare"
+  public
```

Comment:
I've gone two paths with this in my Ruby history.

1) Explicitly putting public/private before every event. This is specifically for explicitness and readability. Unfortunately, tools like rdoc/yard make this hard to maintain, so I do the 2nd one now...
2) Putting `private` at the top of the class definition, and at the very bottom, explicitly public() some methods:

Here's an example of the 2nd option (private by default, explicit public w/ list of methods)
* Private default: https://github.com/jordansissel/ruby-ftw/blob/master/lib/ftw/agent.rb#L56-L58
* Public exposure: https://github.com/jordansissel/ruby-ftw/blob/master/lib/ftw/agent.rb#L460